hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -7,21 +7,212 @@
   import_testplans: ["hw/dv/tools/testplans/csr_testplan.hjson",
                      "hw/dv/tools/testplans/mem_testplan.hjson",
                      "hw/dv/tools/testplans/intr_test_testplan.hjson",
-                     "hw/dv/tools/testplans/tl_device_access_types_testplan.hjson"]
+                     "hw/dv/tools/testplans/tl_device_access_types_testplan.hjson",
+                     "hw/dv/tools/testplans/stress_all_with_reset_testplan.hjson"]
   entries: [
     {
       name: sanity
       desc: '''
-            Basic sanity test acessing a major datapath within the flash_ctrl.
-
-            **Stimulus**:
-            - TBD
-
-            **Checks**:
-            - TBD
+            Randomly read, program or erase (page or a bank) a randomized chunk of flash memory.
+            Only the data partition is accessed. No extra features enabled. Flash memory is
+            invalidated and the targeted chunk is initialized with random data for reads and all 1s
+            for writes. Interrupts are not enabled, Completion is ascertained through polling. The
```

Comment:
is the flash initialized for random data for erase also?

Reply:
No, it is actually [invalidated](https://github.com/lowRISC/opentitan/blob/44f461cad738c8155c9cb98966c7744a146734f7/hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_rand_ops_vseq.sv#L243) (Xs). I can change it to randomly pick beween randomized data and Xs. Let me know if you have any preference here.

Reply:
o okay that's fine too. Just wanted to make sure it had some kind of initial state. 
Btw, have you given any thought as to how these tests might change if we have the real flash underneath instead of the fake one?

Reply:
A little bit, yes. I have a randomization knob to decide whether to check the contents of the memory against the expected via backdoor or via frontdoor (host reads or flash ctrl initiated reads). When we have the real flash, I will disable the backdoor read-check completely.

Reply:
sounds good. We might need to do some path hiding in order to backdoor load the flash, but we can figure that out later.